lib/matplotlib/tests/test_axes.py

```diff
@@ -8455,6 +8455,43 @@ def test_zorder_and_explicit_rasterization():
         fig.savefig(b, format='pdf')
 
 
+@image_comparison(["preset_clip_paths.png"], remove_text=True, style="mpl20")
+def test_preset_clip_paths():
```

Comment:
I'm not sure we want to add an image here that we know is wrong just to have to change it again later?

Some possible alternatives:
* Adding all of the lines/patches completely outside of the rectangle so they are completely clipped? This should make it simpler to compare against a plain rectangle in a check_figures_equal comparison.
* Removing the annotation for now and adding one outside the axes later once that gets fixed.
* Use a Cairo backend if the text clipping works properly there...
* Perhaps simplest though, can you check the expected clip_path after you set it with `artists.get_clip_path() is axes.patch`, `artist.get_clip_path() is polygon` for the various cases.

Reply:
Comparing `get_clip_path` doesn't seem to work:
```
 assert <matplotlib.transforms.TransformedPatchPath object at 0x000002BE4A717F70> is <matplotlib.patches.Polygon object at 0x000002BE4A5465E0>
```

removing annotation and then adding it doesn't really seem better? Now, the test will fail once it is fixed so one will notice this and automatically have to update it (big risk that this test will not be updated.

For `patheffects`, this happens at draw time, so that will not be tested (not 100% sure that it is now as I do not fully understand that part).

I'd take it that one actually would like to take the intersection of e.g. the axes patch and the provided clip path (although I am not sure if that will ever become a problem). So even if the `get_clip_path` approach did work, it would break if this was fixed.

I think this is an example that shows that it is better to use the intersection:
```
import matplotlib as mpl
from matplotlib import pyplot as plt

fig, ax = plt.subplots()
poly3 = mpl.patches.Polygon([[-0.5, 0], [-0.5, 0.5], [0.5, 0.5], [0.5, 0]],
                           facecolor="g", edgecolor="y", linewidth=2, alpha=0.3)

fig.add_artist(poly3, clip=True)

line = mpl.lines.Line2D((-1, 1), (0.25, 0.25), color='r', clip_on=True, clip_path=poly3)
ax.add_artist(line)
# or
# fig.add_artist(line, clip=True)
```

With this PR one gets
![image](https://user-images.githubusercontent.com/8114497/222954956-039a40bf-f1d7-44f8-928d-18d5aa525bbe.png)

while current master gives

![image](https://user-images.githubusercontent.com/8114497/222954975-34e85762-fcaf-45df-a987-6be076eeea2a.png)

or 

![image](https://user-images.githubusercontent.com/8114497/222954984-80dd8300-5a3f-4782-a3d6-026c5b4f4c25.png)

Reply:
I'm not sure I totally follow these recent images... The first one looks like the Polygon isn't actually containing the entire line which I would have expected from the clip, so is there still an error present?

The images from `main` seem to show that the default clip paths are the axes and figure respectively which I expected.

> Comparing get_clip_path doesn't seem to work:

Ahh, yes, if you look at the `def set_clip_path` implementation, I think you can just call `TransformedPathPatch(artist)`?

Sorry for sending you down this rabbit hole :)